lnwallet/channel.go

```diff
@@ -1692,27 +1692,17 @@ func (lc *LightningChannel) restoreStateLogs(
 	pendingRemoteCommitDiff *channeldb.CommitDiff,
 	pendingRemoteKeys *CommitmentKeyRing) error {
 
-	// For each HTLC within the local commitment, we add it to the relevant
-	// update logc based on if it's incoming vs outgoing. For any incoming
-	// HTLC's, we also re-add it to the rHashMap so we can quickly look it
-	// up.
+	// For each incoming HTLC within the local commitment, we add it to the
```

Comment:
Makes sense, nice simplification 👍 

This may also have lend to redundant HTLCs being added to the log, which may cause issues when we go to decide which HTLCs are locked in and ready for forwarding.

Reply:
I _think_ the code I removed would be just noops in all cases. I tried coming up with situations where they would actually add something, but was convinced that they will never by the explanation in the commit message. I might be wrong, but the commit is made by the assumption that it doesn't change behaviour.

Reply:
Thought about this a bit over the past few days, and my current conclusion is that you are indeed correct. Also that these should have technically been no-ops, but could have times caused redundant elements in the linked list.